api/ccapi/include/tensor_dim.h

```diff
@@ -73,15 +79,35 @@ class TensorDim {
      */
     DataType data_type;
 
+    /**
+     * @brief     Data Storage Order : Default is Row-major
+     */
+    StorageOrder storage_order;
+
     /**
      * @brief     Default creator of Tensor Type
      */
-    TensorType() : format(Format::NCHW), data_type(DataType::FP32){};
+    TensorType() :
+      format(Format::NCHW),
+      data_type(DataType::FP32),
+      storage_order(StorageOrder::ROW_MAJOR){};
 
     /**
      * @brief     Default creator of Tensor Type with Format & DataType
      */
-    TensorType(Format fm, DataType d_type) : format(fm), data_type(d_type){};
+    TensorType(Format fm, DataType d_type) :
+      format(fm),
+      data_type(d_type),
+      storage_order(StorageOrder::ROW_MAJOR){};
+
+    /**
+     * @brief     Default creator of Tensor Type with Format, DataType &
+     * StorageOrder
+     */
+    TensorType(Format fm, DataType d_type, StorageOrder order) :
```

Comment:
`TensorType(Format fm, DataType d_type, StorageOrder order = StorageOrder::ROW_MAJOR)`

It would be enough.

Reply:
makes sense :)